envoy/network/socket.h

```diff
@@ -153,7 +178,7 @@ class Socket {
    */
   virtual SocketAddressSetter& addressProvider() PURE;
   virtual const SocketAddressProvider& addressProvider() const PURE;
-  virtual SocketAddressProviderSharedPtr addressProviderSharedPtr() const PURE;
+  virtual SocketAddressSetterSharedPtr addressProviderSharedPtr() const PURE;
```

Comment:
Returning the AddressSetter should only be done from a non-const method. I think the better change is instead of changing this method, add another one `virtual SocketAddressSetterSharedPtr addressProviderSharedPtr() PURE;` (non-const version).

Reply:
got it, thanks!

Reply:
If we keep the upstream ssl info in the streamInfo, then we needn't change this interface anymore